test/test_tensordict.py

```diff
@@ -2479,6 +2479,12 @@ def test_tensordict_device(self, device):
         assert tensordict["c"].device == device
 
 
+@pytest.mark.parametrize("separator", [".", "-"])
+def test_unflatten_keys_collision(separator):
+    td = TensorDict({"a": [1, 2], f"c{separator}a": [1, 2], "c": TensorDict({"b": [1, 2]}, [])}, [])
+    ref = TensorDict({"a": [1, 2], "c": TensorDict({"a": [1, 2], "b": [1, 2]}, [])}, [])
+    assert repr(td.unflatten_keys(separator)) == repr(ref)
```

Comment:
Comparing the results of `repr` feels like it might be a bit brittle. Might be better to do something like

```python
from tensordict.tensordict import assert_allclose_td

assert_allclose_td(td.unflatten_keys(separator), ref)
```

Reply:
Got it.